src/java.base/share/classes/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java

```diff
@@ -456,7 +455,7 @@ private static boolean isAncestor(ClassLoader first, ClassLoader second) {
          */
         private static boolean isSamePackage(Class<?> class1, Class<?> class2) {
             return class1.getClassLoader() == class2.getClassLoader()
-                   && Objects.equals(class1.getPackageName(), class2.getPackageName());
+                   && class1.getPackageName().equals(class2.getPackageName());
```

Comment:
If the j.u.c.atomic code is changed then it will need to be coordinated with Doug Lea's repo. The rest of the changes are good and thanks for following up from the comments in the other issue.

Reply:
Hello, how should I do it? Maybe we can just mention him here?

Reply:
@DougLea could you have a look into the part of this PR related to java.util.concurrent package?

Reply:
@AlanBateman if DL is not responding, will it be ok to just revert the changes related to `java.util.concurrent`?

Reply:
That should be fine, the null check in Objects.equals is benign with these usages.

Reply:
One more thing I'm thinking about (not to be done in this PR of course) is to move call to `String.intern()` from where it is now in `Class.getPackageName()`
```java
public String getPackageName() {
    String pn = this.packageName;
    if (pn == null) {
        Class<?> c = this;
        while (c.isArray()) {
            c = c.getComponentType();
        }
        if (c.isPrimitive()) {
            pn = "java.lang";
        } else {
            String cn = c.getName();
            int dot = cn.lastIndexOf('.');
            pn = (dot != -1) ? cn.substring(0, dot).intern() : "";    // <---
        }
        this.packageName = pn;
    }
    return pn;
}
```
to `packageName` field assignement like
```java
this.packageName = pn.intern();
```
this would add two more Strings (`""` and `"java.lang"`) into string table and allow to avoid `String.equals()` in favour of `==` call when comparing package names. What do you think?

Reply:
String literals (like "") are implicitly interned:
```		System.out.println("" == new String("")); // false
		System.out.println("" == new String("").intern()); // true
		System.out.println("java.lang" == new String("java.lang")); // false
		System.out.println("java.lang" == new String("java.lang").intern()); // true```

This means we could already use `==`  instead of `equals` when evaluating the result of `getPackageName`.

Reply:
Done!